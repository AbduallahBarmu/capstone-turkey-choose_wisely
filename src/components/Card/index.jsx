import React from "react";
import { Row, Col, Container } from "react-bootstrap";
import "./Style.scss";
import Avatar from "@material-ui/core/Avatar";

export default function SliderSecion(speech) {
  console.log(speech);

  return (
    <Container>
      <Row className="align-items-center justify-content-center Card">
        <Col className="ps-5 avatarComponent" md={2}>
          <Avatar className="Avatar" alt="Remy Sharp" />
        </Col>

        <Col className="col-md-offset-2 SpeechComponent" md={10}>
          <p className="paragraph">{speech.part1}</p>
          <h5 className="auther">{speech.part2}</h5>
          <h6 className="description">{speech.part3}</h6>
        </Col>
      </Row>
    </Container>
  );
}
