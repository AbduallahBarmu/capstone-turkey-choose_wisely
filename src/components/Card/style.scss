@import "../../style/mixin.scss";
.Card {
  background-color: #fff;
  padding-left: 1.071875rem;
  padding-right: 1.4975rem;
  padding-top: 4.080625rem;
  padding-bottom: 0.6875rem;
  transition: scale 0.5s ease-in-out;
  .SpeechComponent {
    padding-left: 6.71875rem;
    padding-bottom: 0.765625rem;
    .paragraph {
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 1.125rem;

      letter-spacing: -0.03125rem;
      color: #386641;
      @include for-desktop-up {
        width: 70%;
      }
    }

    .auther {
      height: 1.0625rem;
      font-family: Roboto;
      font-style: normal;
      font-weight: bold;
      font-size: 0.875rem;
      line-height: 1rem;
      letter-spacing: -0.019886375rem;
      color: #666666;
    }
    .description {
      height: 0.9375rem;
      font-family: Roboto;
      font-style: normal;
      font-weight: 300;
      font-size: 0.75rem;
      line-height: 0.875rem;
      letter-spacing: -0.0170454375rem;

      color: #666666;
    }
  }
  .avatarComponent {
    padding-right: 4.461875rem;
    .Avatar {
      width: 10.3125rem;
      height: 10.3125rem;
      background: #eef2f4;
    }
  }
}
