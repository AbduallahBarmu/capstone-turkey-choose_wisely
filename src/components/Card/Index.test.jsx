import React from "react";
import Card from "./Index";
import renderer from "react-test-renderer";

const speech =  {
    "part1": "Currently, Ford sells six sedans and coupes in North America with the Fiesta, Focus, Fusion, C-Max, Mustang and Taurus. This lineup hits multiple segments from the compact Fiest to the mid-size Focus, C-Max and Fusion to the full-size Taurus",
    "part2": " Emily JÃ¸rgense",
    "part3": " Author at Panoply Store"
  } 
test("Snapshop test of Card component", () => {
  const map = renderer.create(<Card {...speech}/>);
  let tree = map.toJSON();

  expect(tree).toMatchSnapshot();
});