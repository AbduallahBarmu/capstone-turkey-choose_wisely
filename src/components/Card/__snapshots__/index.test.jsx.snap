// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop test of Card component 1`] = `
<div
  className="container"
>
  <div
    className="align-items-center justify-content-center Card row"
  >
    <div
      className="ps-5 avatarComponent col-md-2"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular Avatar MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="col-md-offset-2 SpeechComponent col-md-10"
    >
      <p
        className="paragraph"
      >
        Currently, Ford sells six sedans and coupes in North America with the Fiesta, Focus, Fusion, C-Max, Mustang and Taurus. This lineup hits multiple segments from the compact Fiest to the mid-size Focus, C-Max and Fusion to the full-size Taurus
      </p>
      <h5
        className="auther"
      >
         Emily JÃ¸rgense
      </h5>
      <h6
        className="description"
      >
         Author at Panoply Store
      </h6>
    </div>
  </div>
</div>
`;
